pkgs/misc/uboot/default.nix

```diff
@@ -424,7 +418,6 @@ in {
   };
```

Comment:
Is the proprietary ram training still required here? I have no idea if the Rock64 situation improved since. (And don't have the hardware to test.)

* * *

Can the SPI image be built, like it is now for RockPro64?

Reply:
> Is the proprietary ram training still required here? I have no idea if the Rock64 situation improved since. (And don't have the hardware to test.)

I'm not sure. I have booted with the upstream TPL while trying to get the SPI image working, but only as far as running U-Boot (didn't boot Linux). I can try looking at this, would it just be using upstream TPL and leaving the board running to see if anything crashes?

> Can the SPI image be built, like it is now for RockPro64?

I can build it in the same way, but after I flash it the board acts as though it doesn't detect U-Boot in the SPI flash and boots from eMMC/SD card (or has no output if neither are present, presumably never leaving the boot ROM). So I decided to just put up the RockPro64 one for now since it worked, and investigate Rock64 later. Figuring that out is outside my current knowledge at the moment, I'm not sure where to begin.

Reply:
If you are able to boot and log in, that probably indicates that the problem is fixed, since it used to crash during boot. My Rock64 seems to have hardware problems, so I wasn't able to test #145506

Reply:
I didn't try booting all the way to Linux. I'll try that now and comment on #145506.

Reply:
It worked fine with the U-Boot from #145506. I'm happy to have that merged and rebase this PR.

Reply:
I believe if you cherry-pick it into your PRit would keep author information intact and be okay; "even better" if you cherry-pick it "before" your own changes I suppose(?)

(I also left notes about a different platform's instability with OSS ram training, if it can help test things out...)

Reply:
That does work, I've cherry-picked the commit.